static/js/src/advantage/contracts-api.js

```diff
@@ -306,3 +306,82 @@ export async function setAutoRenewal(value) {
   let data = await response.json();
   return data;
 }
+
+export async function postGuestFreeTrial({
+  email,
+  account_name,
+  name,
+  address,
+  productListingId,
+  quantity,
+}) {
+  const queryString = window.location.search; // Pass arguments to the flask backend eg. "test=backend=true"
+
+  let response = await fetch(`/advantage/post_guest_trial${queryString}`, {
+    method: "POST",
+    cache: "no-store",
+    credentials: "include",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      email: email,
+      account_name: account_name,
+      name: name,
+      address: address,
+      products: [
+        {
+          product_listing_id: productListingId,
+          quantity: quantity,
+        },
+      ],
+    }),
+  });
+
+  if (response.ok) {
```

Comment:
I'm not sure that this response handling is valuable in the API as it's written. It's actually reducing the amount of data that the consumer is receiving but it's not providing any protection against the failure cases or if the json fails to parse. 

If you wanted to just return the data from the request then I might suggest a structure like...

`return { data, err };`

So you'd only ever return the successfully parsed data and an error message that would default to null so the consumer would not have to wrap this function in a try/catch

So it would take the usage from...

```js
let response = null;
try {
 response = await postGuestFreeTrial(...);
catch (e) {
 // handle error
}
// use response
```

to...

```js

const {data, err} = await postGuestFreeTrial(...);
if (err) {
 // handle error
}
//use data
```

I'd apply similar changes to the other function in this file as well.